web/cobrands/fixmystreet/assets.js

```diff
@@ -426,12 +421,24 @@ fixmystreet.assets = {
             return hide_assets;
         })(fixmystreet.maps.display_around);
 
+        var pins_layer = fixmystreet.map.getLayersByName("Pins")[0];
         for (var i = 0; i < fixmystreet.assets.layers.length; i++) {
             var layer = fixmystreet.assets.layers[i];
             fixmystreet.map.addLayer(layer);
             if (layer.fixmystreet.asset_category) {
                 fixmystreet.map.events.register( 'zoomend', layer, check_zoom_message_visibility);
             }
+
+            // Don't cover the existing pins layer
+            if (pins_layer) {
+                layer.setZIndex(pins_layer.getZIndex()-1);
+            }
+
+            // Make sure the fault markers always appear beneath the linked assets
+            if (layer.fixmystreet.fault_layer) {
+                layer.fixmystreet.fault_layer.setZIndex(layer.getZIndex()-1);
```

Comment:
We're in a loop here through all the layers, both assets and faults. So tis won't work (I think) because it'll do the asset layer first (set z to pins-1, set faults to assets-1), then it'll do the fault layer and set z to pins-1 again?

Reply:
Yep, you're right. Will rejig this to handle that.